projects/packages/jetpack-mu-wpcom/src/features/wpcom-block-editor/class-jetpack-wpcom-block-editor.php

```diff
@@ -0,0 +1,630 @@
+	 * @param string $nonce Nonce that was used in the form to verify.
+	 * @param string $action Should give context to what is taking place and be the same when nonce was created.
+	 * @return boolean|WP_Error Whether the nonce is valid.
+	 */
+	public function verify_frame_nonce( $nonce, $action ) {
+		if ( empty( $nonce ) ) {
+			return false;
+		}
+
+		list( $expiration, $user_id, $hash ) = explode( ':', $nonce, 3 );
+
+		$this->nonce_user_id = (int) $user_id;
+		if ( ! $this->nonce_user_id ) {
+			return false;
+		}
+
+		$token = ( new Tokens() )->get_access_token( $this->nonce_user_id );
+		if ( ! $token ) {
+			return false;
+		}
+
+		/*
+		 * Failures must return `false` (blocking the iframe) prior to the
+		 * signature verification.
+		 */
+
+		add_filter( 'salt', array( $this, 'filter_salt' ), 10, 2 );
+		$expected_hash = wp_hash( "$expiration|$action|{$this->nonce_user_id}", 'jetpack_frame_nonce' );
+		remove_filter( 'salt', array( $this, 'filter_salt' ) );
+
+		if ( ! hash_equals( $hash, $expected_hash ) ) {
+			return false;
+		}
+
+		/*
+		 * Failures may return `WP_Error` (showing an error in the iframe) after the
+		 * signature verification passes.
+		 */
+
+		if ( time() > $expiration ) {
+			return new WP_Error( 'nonce_invalid_expired', 'Expired nonce.', array( 'status' => 401 ) );
+		}
+
+		// Check if it matches the current user, unless they're trying to log in.
+		if ( get_current_user_id() !== $this->nonce_user_id && ! doing_action( 'login_init' ) ) {
+			return new WP_Error( 'nonce_invalid_user_mismatch', 'User ID mismatch.', array( 'status' => 401 ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Filters the WordPress salt.
+	 *
+	 * @param string $salt Salt for the given scheme.
+	 * @param string $scheme Authentication scheme.
+	 * @return string
+	 */
+	public function filter_salt( $salt, $scheme ) {
+		if ( 'jetpack_frame_nonce' === $scheme ) {
+			$token = ( new Tokens() )->get_access_token( $this->nonce_user_id );
+
+			if ( $token ) {
+				$salt = $token->secret;
+			}
+		}
+
+		return $salt;
+	}
+
+	/**
+	 * Enqueues the WordPress.com block editor integration assets for the editor.
+	 */
+	public function enqueue_block_editor_assets() {
+		global $pagenow;
+
+		// Bail if we're not in the post editor, but on the widget settings screen.
+		if ( is_customize_preview() || 'widgets.php' === $pagenow ) {
+			return;
+		}
+
+		$debug   = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG;
+		$version = gmdate( 'Ymd' );
+
+		wp_enqueue_script(
+			'wpcom-block-editor-default-editor-script',
+			$debug
+				? '//widgets.wp.com/wpcom-block-editor/default.editor.js?minify=false'
+				: '//widgets.wp.com/wpcom-block-editor/default.editor.min.js',
+			array(
+				'jquery',
+				'lodash',
+				'wp-annotations',
+				'wp-compose',
+				'wp-data',
+				'wp-editor',
+				'wp-element',
+				'wp-rich-text',
+			),
+			$version,
+			true
+		);
+
+		wp_localize_script(
+			'wpcom-block-editor-default-editor-script',
+			'jetpackMuWpcomGutenberg',
```

Comment:
I wonder if we could change the variable back to `wpcomGutenberg` from `jetpackMuWpcomGutenberg`, but only on Atomic sites, as I know that it was changed because of a Simple issue: https://github.com/Automattic/jetpack/pull/37324#issuecomment-2124666486